build.sh

```diff
@@ -356,117 +491,187 @@ EOF
-        ;;
-        --noconfig)
-            ALLOW_KERNEL_CONFIG_CHANGES='no'
-            shift;
-        ;;
-
-        --kernelonly)
-            BUILD_ROOT='no'
-            BUILD_INITRAMFS='no'
-            shift;
-        ;;
-        --rootonly)
-            BUILD_KERNEL='no'
-            BUILD_INITRAMFS='no'
-            shift;
-        ;;
-        --nokernel)
-            BUILD_KERNEL='no'
-            shift;
-        ;;
-        --noinitramfs)
-            BUILD_INITRAMFS='no'
-            shift;
-        ;;
-
-        --changes)
-            ALLOW_ROOTFS_CHANGES='yes'
-            shift;
-        ;;
-	--nosudo)
-	    NO_SUDO='yes'
-	    shift;
-	;;
-
-	--ghrunner)
-	    BUILD_KERNEL='yes'
-	    BUILD_INITRAMFS='no'
-	    BUILD_ROOT='no'
-	    GHRUNNER='yes'
-	    THREADS=2
-	    ALLOW_KERNEL_CONFIG_CHANGES='no'
-	    shift;
-	;;
-        *)    # unknown option
-            POSITIONAL+=("$1") # save it in an array for later
-            shift # past argument
-        ;;
-    esac
-done
-
-if [[ "${EUID}" != "0" ]] && [[ $GHRUNNER != 'yes' ]]; then
+# Find dir of build.sh and go there
+current_dir="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
+# check for whitespace in ${current_dir} and exit for safety reasons
+grep -q "[[:space:]]" <<<"${current_dir}" && { echo "\"${current_dir}\" contains whitespace. Aborting." >&2 ; exit 1 ; }
+cd "${current_dir}" || exit
+
+DIALOG_CANCEL=1
+DIALOG_ESC=255
+BACKTITLE="WDMC kernel & rootfs build script"
+
+exec 3>&1
+dsize=$(dialog --print-maxsize 2>&1 1>&3)
+exec 3>&-
+dsize=( $(echo "$dsize" | grep -o -E '[0-9]+') )
+DIALOG_WIDTH=$((${dsize[0]} / 2))
+DIALOG_HEIGHT=$((${dsize[1]} / 2))
+
+
+############################################################
+### (SUB) DIRECTORIES TO USE
+############################################################
+output_dir="${current_dir}/output"
+rootfs_dir="${output_dir}/rootfs"
+boot_dir="${output_dir}/boot"
+cache_dir="${current_dir}/cache"
+
+current_user="$(stat --format %U "${current_dir}"/.git)"
+
+############################################################
+### Begin figuring out configuration
+############################################################
+GHRUNNER='off'
+THREADS=8
+EXTRA_PKGS='bash-completion htop'
+makehelp='make CROSS_COMPILE=/usr/bin/arm-none-eabi- ARCH=arm'                                         #FOR KERNEL VERSION >= 5.6 (via apt)
+
+# start by reading command line arguments
+read_arguments "$@"
+
+if [[ "${EUID}" != "0" ]] && [[ $GHRUNNER != 'on' ]]; then
     echo "This script requires root privileges, please rerun using sudo"
     exit 1
 fi
 
-echo '### Build options'
-
-echo "Build dir: ${current_dir}"
+if [[ $GHRUNNER != 'on' ]]; then
+    echo "### Will try to use apt to install prerequisites."
+    apt-get install build-essential bc libncurses5 u-boot-tools git libncurses-dev lib32z1 lib32ncurses5-dev libmpc-dev libmpfr-dev libgmp3-dev flex bison debootstrap debian-archive-keyring qemu-user-static gcc-arm-none-eabi
```

Comment:
I would suggest adding dialog, because for example my wsl2 Ubuntu 20.04 didn't include dialog and therefore couldn't create the dialog in the cli window.

```suggestion
    apt-get install build-essential bc libncurses5 dialog u-boot-tools git libncurses-dev lib32z1 lib32ncurses5-dev libmpc-dev libmpfr-dev libgmp3-dev flex bison debootstrap debian-archive-keyring qemu-user-static gcc-arm-none-eabi
```